core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -227,8 +229,8 @@ else if (newState.isDone()) {
                 }
             }
 
-            // notify that task state changed (apart from initial RUNNING state notification)
-            if (newState != RUNNING) {
+            // notify that task state changed (apart from initial INITIALIZING state notification)
+            if (newState != INITIALIZING) {
```

Comment:
do we want to notify about `RUNNING` now?

Reply:
good question, if notifying useful for terminating states only then probably not?

Reply:
if we don't care about notification from changing state from `INITIALIZING -> RUNNING`, then why an extra state is needed?

Reply:
Extra state is needed to block execution of the query before all catalogs are loaded on the worker.

Reply:
The reasoning for not notifying on certain changes is not something I full get however